### PR TITLE
Ddf 1790 - Changes SystemBaseUrl and SystemInfo to be static utilties

### PR DESCRIPTION
**Pull request by coyotesqrl**
`SystemBaseUrl` had incorrectly been passed between bundles and is best implemented as a static utility class; in the process of working on this issue, `SystemInfo` was noted to be used as an instance class as well when it was better implemented as a static utility class.

@rzwiefel 
@tbatie 
@ryeats 
@stustison

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/458" target="_blank">ReviewNinja

**Comment by ryeats** [comment]
:+1:

File: catalog/transformer/catalog-transformer-service-atom/src/test/java/ddf/catalog/transformer/response/query/atom/TestAtomTransformer.java [hunk 1]
```diff
@@ -187,34 +187,41 @@ public void testNoDdfConfiguration()
...
-        transformer.setMetacardTransformer(metacardTransformer);
-        SourceResponse response = getSourceResponseStub(SAMPLE_ID, null);
-
-        // when
-        BinaryContent binaryContent = transformer.transform(response, null);
-
-        // then
-        assertThat(binaryContent.getMimeType(), is(AtomTransformer.MIME_TYPE));
+        try {
+            System.setProperty(SystemInfo.SITE_NAME, "");
```
Comment (lessarderic) [inline catalog/transformer/catalog-transformer-service-atom/src/test/java/ddf/catalog/transformer/response/query/atom/TestAtomTransformer.java]: This is the concern I have with this change, i.e., the tests cannot mock `SystemInfo` and `SystemBaseUrl` anymore and have to know about their implementation to actually have them return specific values.

Comment (coyotesqrl) [inline catalog/transformer/catalog-transformer-service-atom/src/test/java/ddf/catalog/transformer/response/query/atom/TestAtomTransformer.java]: In general I agree. However for this case, we're trying to limit the problems that can occur if someone were to share instances of these classes between bundles. As they're simple utility facades, I think the benefit of reducing risk is greater than the testing downside in this case.

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/CatalogFrameworkImpl.java [hunk 2]
```diff
@@ -2115,8 +2115,8 @@ public SystemInfo getSystemInfo() {
```
Comment (stustison) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/CatalogFrameworkImpl.java]: I removed this file, might need to rebase.

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/util/DdfConfigurationManager.java [hunk 3]
```diff
@@ -157,8 +157,6 @@
```
Comment (stustison) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/util/DdfConfigurationManager.java]: Same here, this file is gone.

File: platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java [hunk 4]
```diff
@@ -382,7 +380,7 @@ private Response getErrorResponse(String relayState, AuthnRequest authnRequest,
             throws WSSecurityException, IOException, SimpleSign.SignatureException {
         LOGGER.debug("Creating SAML Response for error condition.");
         org.opensaml.saml.saml2.core.Response samlResponse = SamlProtocol.createResponse(
-                SamlProtocol.createIssuer(systemBaseUrl.constructUrl("/idp/login", true)),
+                SamlProtocol.createIssuer(SystemBaseUrl.constructUrl("/idp/login", true)),
```
Comment (tbatie) [inline platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java]: I noticed the pom isn't embedding common-system. Won't this fail to find SystemBaseUrl at runtime?

File: platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/KarafLogoutAction.java [hunk 5]
```diff
@@ -39,10 +39,10 @@
 
     static {
         try {
-            logoutUrl = new URL(new SystemBaseUrl().constructUrl("/logout/local"));
+            logoutUrl = new URL(SystemBaseUrl.constructUrl("/logout/local"));
```
Comment (tbatie) [inline platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/KarafLogoutAction.java]: Same question here with common-system not being embedded in the pom

File: platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrServerFactory.java [hunk 6]
```diff
@@ -119,11 +117,11 @@ private static ExecutorService getThreadPool() {
     }
 
     public static String getDefaultHttpsAddress() {
-        return systemBaseUrl.constructUrl("https", "/solr");
+        return SystemBaseUrl.constructUrl("https", "/solr");
```
Comment (tbatie) [inline platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrServerFactory.java]: Same question here with common-system not being embedded in the pom

File: platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java (hunk 4 above)
Comment (coyotesqrl) [inline platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java]: Good catch. It looks like it was a rebase issue. I'll fix it and check the others which are likely all the same.

File: platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrServerFactory.java (hunk 6 above)
Comment (coyotesqrl) [inline platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrServerFactory.java]: This isn't a bundle so it's not a problem.

**Comment by rzwiefel** [comment]
built okay, started up and ran successfully.
[selfie-0](https://github.com/thieman/github-selfies/)

**Pull request merged**